src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -287,10 +287,21 @@ bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(app::CommandH
     DeviceControlServer * server = &DeviceLayer::DeviceControlServer::DeviceControlSvr();
     Commands::SetRegulatoryConfigResponse::Type response;
 
+    auto & countryCode = commandData.countryCode;
+    bool isValidLength = countryCode.size() == DeviceLayer::ConfigurationManager::kMaxLocationLength;
+    if (!isValidLength)
+    {
+        ChipLogError(Zcl, "Invalid country code: '%.*s'", static_cast<int>(countryCode.size()), countryCode.data());
+        commandObj->AddStatus(commandPath, Protocols::InteractionModel::Status::ConstraintError);
+        return true;
+    }
+
     if (commandData.newRegulatoryConfig > RegulatoryLocationTypeEnum::kIndoorOutdoor)
     {
         response.errorCode = CommissioningErrorEnum::kValueOutsideRange;
-        response.debugText = commandData.countryCode;
+        // TODO: How does using the country code in debug text make sense, if
```

Comment:
I guess it depends what we're expecting the commissioner to do with it. I suppose the commissioner could surface something like `Failed to set country code to <new country code>. Country code is set to <previous country code from debug text>.`

But that's just a guess. I don't really know for sure. And either way, the commissioner should either have or can easily get both pieces of information. And I could be really wrong there because per the spec
```
Some response commands have a DebugText argument which SHOULD NOT be presented directly in user interfaces. Its purpose is to help developers in troubleshooting errors. The value MAY go into logs or crash reports.
```

Which is a long-winded way of saying that I'm about as stumped as you are and I'm not clear on the intent for this field. I almost think that "Invalid country code" makes more sense here as a debug text response because it distinguishes that the failure was the country code and not the regulatory location. Either way, not a blocker for this PR.